src/rgw/rgw_common.cc

```diff
@@ -2011,5 +1926,22 @@ string camelcase_dash_http_attr(const string& orig)
         last_sep = false;
     }
   }
-  return string(buf);
+  return buf;
+}
+
+// Make a modified copy:
+string uppercase_dash_http_attr(string s)
+{
+  return transform(begin(s), end(s), begin(s), [](const char c) {
```

Comment:
i'd suggest put `return s` in a separated line instead.

Reply:
While I do prefer the expression-form with the comma operator in this case, I don't feel too strongly about it and it's really just a matter of taste here, so I'll change it. But is there any technical reason you prefer the statement form here?

Reply:
IMO, it hurts the readability if i need to parse the statement with comma operator, and to find the return value at end of it.

Reply:
Sure, and this is in no small part because C and C++ use statements as the primary unit of organization: values and computations are consistently separate things, and the statements tend to be effect-ful. The point of using an expression instead is to focus on the /computed value/ and their inseparability from the each other and the meaning of the function-- it would be ideal to be able to write:
auto f(T xs) { return transform(xs, fn); } // e.g. f xs = map fn xs
...but we can't because transform() returns an iterator rather than the new value. But, the comma operator lets us write what we mean without allowing the computation to be /separate/ from the value: the function has no "statements" and nothing can be introduced. (There are other uses of the comma operator, especially pre- C++17 that are important particularly in metaprogramming, but not germane here.)
Clearly, this is a bit unusual (though I'm hardly the first to do it), but that is the thought behind it. Parsing the whole statement is unfortunate, but arguably no great moment as you read "return" and then the expression (no worse than, say, a ternary return-- but more unusual). Anyway, that's some of the reasoning for that style of return-expression. I think it's purely a matter of aesthetics (and certainly shoving everything into an expression would be unnatural in C++), and the statement version of the function is also trivial so since there was objection I don't mind changing it here (done), but I thought it would be good to briefly discuss.